pinot-common/src/test/java/org/apache/pinot/common/metrics/AbstractMetricsTest.java

```diff
@@ -108,4 +112,154 @@ public void testMultipleGauges() {
     controllerMetrics.removeGauge(metricName2);
     Assert.assertTrue(controllerMetrics.getMetricsRegistry().allMetrics().isEmpty());
   }
+
+  /**
+   * Creates and initializes a concrete instance of {@link AbstractMetrics} (in this case, a {@code ControllerMetrics}).
+   * @return a {@code ControllerMetrics} suitable for testing {@code AbstractMetrics} APIs
+   */
+  private static ControllerMetrics buildTestMetrics() {
```

Comment:
nit: should this just be a class level `ControllerMetrics` object that is created before each test? Or are you trying to leave this parallelizable?

Reply:
The main reason is to ensure each test starts with clean metrics. If all tests shared the same ControllerMetrics object, you could have one test fail because the previous test didn't fully clean the shared metrics. It also, as you mention, makes the tests parallelizable which is nice to have.